Execute the PTO alarm immediately if in the past (#4831)
quicwg/base-drafts: draft-ietf-quic-recovery.md

as specified in {{Section 8.1 of QUIC-TRANSPORT}}. If no additional data can be
sent, the server's PTO timer MUST NOT be armed until datagrams have been
received from the client, because packets sent on PTO count against the
anti-amplification limit.  The server's PTO timer could be re-armed to a time
in the past after being amplification limited, in which case it is executed

> immediately. Doing so avoids sending new packets in ApplicationData prior to

Reviewer: `ApplicationData` is so far only used in the pseudo code. Can we rephrase this?

> critical to the completion of the handshake.  Note that the server could fail

Author: I created a new paragraph and worked in the beginning of your suggestion, but I was having a harder time with the end.

> to validate the client's address even if 0-RTT is accepted.

Author: I reworked it to connect it back, which I think makes sense given this mostly happens when 0-RTT is accepted, but address validation fails.

as specified in {{Section 8.1 of QUIC-TRANSPORT}}. If no additional data can be
sent, the server's PTO timer MUST NOT be armed until datagrams have been
received from the client, because packets sent on PTO count against the
anti-amplification limit.

When the server receives a datagram from the client, the amplification limit is
increased and the server resets the PTO timer.  If the PTO timer is then set to
a time in the past, it is executed immediately. Doing so avoids sending new
1-RTT packets prior to packets critical to the completion of the handshake.

> This could be common when 0-RTT is accepted, but the server fails to validate

Reviewer 2: ```suggestion
This can happen when 0-RTT is accepted but the server fails to validate
```

Author: It can happen without 0-RTT, it's just a lot more likely with 0-RTT.

> This could be common when 0-RTT is accepted, but the server fails to validate
> the client's address.

Reviewer 2: ```suggestion
In particular, this can happen when 0-RTT is accepted but the server fails to
validate the client's address.
```

  // PTO timer to avoid deadlock.
  if (server was at anti-amplification limit):
    SetLossDetectionTimer()
    if loss_detection_timer.timeout < now():
      // Execute PTO if it would have expired

> 	    // while the amplification limit applied.

Author: ```suggestion
	     // while the amplification limit applied.
```
